platforms/unix/vm/sqUnixExternalPrims.c

```diff
@@ -202,11 +203,7 @@ void *ioLoadModule(char *pluginName)
 
   /* try dlopen()ing LIBRARY_PREFIX<name>LIBRARY_SUFFIX searching only the default locations modulo LD_LIBRARY_PATH et al */
 
-# if defined(HAVE_SNPRINTF)
```

Comment:
I agree you on using a portable sprintf where it is not available. That means to things:

1. Those HAVE_SNPRINTF would go away anyways, because instead of being used everywhere in the source code, it will be used in a single place: the Makefiles, for conditionally linking the portable version.

2. I think that platforms we care about support snprintf, so there are no reasons for working around the lack of snprintf (and even more so when, as you pointed out, we can use a portable version)